bson/json_util.py

```diff
@@ -855,100 +855,189 @@ def _encode_int64(obj: Int64, json_options: JSONOptions) -> Any:
+        if not obj.tzinfo:
+            obj = obj.replace(tzinfo=utc)
+            assert obj.tzinfo is not None
+        if obj >= EPOCH_AWARE:
+            off = obj.tzinfo.utcoffset(obj)
+            if (off.days, off.seconds, off.microseconds) == (0, 0, 0):  # type: ignore
+                tz_string = "Z"
+            else:
+                tz_string = obj.strftime("%z")
+            millis = int(obj.microsecond / 1000)
+            fracsecs = ".%03d" % (millis,) if millis else ""
+            return {
+                "$date": "{}{}{}".format(obj.strftime("%Y-%m-%dT%H:%M:%S"), fracsecs, tz_string)
+            }
+
+    millis = _datetime_to_millis(obj)
+    if json_options.datetime_representation == DatetimeRepresentation.LEGACY:
+        return {"$date": millis}
+    return {"$date": {"$numberLong": str(millis)}}
+
+
+def _encode_bytes(obj: bytes, json_options: JSONOptions) -> dict:
+    return _encode_binary(obj, 0, json_options)
+
+
+def _encode_binary_obj(obj: Binary, json_options: JSONOptions) -> dict:
+    return _encode_binary(obj, obj.subtype, json_options)
+
+
+def _encode_uuid(obj: uuid.UUID, json_options: JSONOptions) -> dict:
+    if json_options.strict_uuid:
+        binval = Binary.from_uuid(obj, uuid_representation=json_options.uuid_representation)
+        return _encode_binary(binval, binval.subtype, json_options)
+    else:
+        return {"$uuid": obj.hex}
+
+
+def _encode_objectid(obj: ObjectId, dummy0: Any) -> dict:
+    return {"$oid": str(obj)}
+
+
+def _encode_timestamp(obj: Timestamp, dummy0: Any) -> dict:
+    return {"$timestamp": {"t": obj.time, "i": obj.inc}}
+
+
+def _encode_decimal128(obj: Timestamp, dummy0: Any) -> dict:
+    return {"$numberDecimal": str(obj)}
+
+
+def _encode_dbref(obj: DBRef, json_options: JSONOptions) -> dict:
+    return _json_convert(obj.as_doc(), json_options=json_options)
+
+
+def _encode_minkey(dummy0: Any, dummy1: Any) -> dict:
+    return {"$minKey": 1}
+
+
+def _encode_maxkey(dummy0: Any, dummy1: Any) -> dict:
+    return {"$maxKey": 1}
+
+
 # Encoders for BSON types
-_encoders = {
-    5: lambda obj, json_options: _encode_binary(obj, obj.subtype, json_options),  # Binary
-    7: lambda obj, json_options: {"$oid": str(obj)},  # noqa: ARG005 ObjectId
-    9: _encode_datetimems,  # DatetimeMS
-    13: _encode_code,  # Code
-    17: lambda obj, json_options: {"$timestamp": {"t": obj.time, "i": obj.inc}},  # noqa: ARG005 Timestamp
-    18: _encode_int64,  # Int64
-    19: lambda obj, json_options: {"$numberDecimal": str(obj)},  # noqa: ARG005 Decimal128
-    100: lambda obj, json_options: _json_convert(obj.as_doc(), json_options=json_options),  # DBRef
-    127: lambda obj, json_options: {"$maxKey": 1},  # noqa: ARG005 MaxKey
-    255: lambda obj, json_options: {"$minKey": 1},  # noqa: ARG005 MinKey
+# Each encoder function's signature is:
+#   - obj: a Python data type, e.g. a Python int for _encode_int
+#   - json_options: a JSONOptions
+_ENCODERS: dict[Any, Any] = {
```

Comment:
Is it worth type-hinting the value type here since we know all the function signatures?

Reply:
Done.